drivers/ata/ahciem.c

```diff
@@ -436,6 +428,13 @@ static int ahciem_queuecommand(struct Scsi_Host *shost, struct scsi_cmnd *cmd)
 	struct ahciem_args args = { .cmd = cmd, .enc = enc };
 	const u8 *cdb = cmd->cmnd;
 
+	if (unlikely(!cmd->cmd_len)) {
+		DPRINTK("bad CDB len for scsi op 0x%02x\n", cdb[0]);
+		cmd->result = DID_ERROR << 16;
+		cmd->scsi_done(cmd);
+		return 0;
+	}
+
```

Comment:
I see where it was taken from, but it looks weird to me to check for CDB length of zero and then access first CDB byte. ;)